cmd/microcloud/main_init.go

```diff
@@ -763,10 +763,16 @@ func (c *initConfig) setupCluster(s *service.Handler) error {
 				}
 
 				logger.Debug("Adding disk to MicroCeph", logger.Ctx{"name": name, "disk": disk.Path})
-				_, err = cephClient.AddDisk(context.Background(), client, &disk)
+				resp, err := cephClient.AddDisk(context.Background(), client, &disk)
 				if err != nil {
 					return err
 				}
+
+				for _, report := range resp.Reports {
+					if report.Error != "" {
+						return fmt.Errorf("Failed to add disk to MicroCeph: %s", report.Error)
```

Comment:
Do we have to concatenate the errors from multiple `Reports`?

Reply:
Fixed.